SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Build/Build.cs

```diff
@@ -89,17 +89,33 @@ public static void Main(string[] args)
             // Locate the Unreal Engine.
             Console.WriteLine("Finding Unreal Engine build.");
             string uproject = File.ReadAllText(projectFile, Encoding.UTF8);
-            string engineAssociationRegex = "..{8}-.{4}-.{4}-.{4}-.{12}.";
-            Match engineAssociationMatch = Regex.Match(uproject, engineAssociationRegex);
 
             string unrealEngine = "";
+            string engineAssociation = "";
 
-            // If we found an engine association within the .uproject, query the registry for the engine directory.
-            if (engineAssociationMatch.Success)
-            {
-                string engineAssociation = engineAssociationMatch.Value;
-                Console.WriteLine("Engine Association: " + engineAssociation);
+            dynamic projectJson = JObject.Parse(uproject);
+            dynamic assocation = projectJson.EngineAssociation;
```

Comment:
I think you can collapse the assignments down even more

Reply:
True!